enterprise/internal/batches/reconciler/executor.go

```
@@ -427,6 +457,21 @@ func loadChangesetSource(ctx context.Context, s *store.Store, sourcer sources.So
 	return css, nil
 }
 
+var errChangesetSourceCannotFork = errors.New("forking is enabled, but the changeset source does not support forks")
+
+func loadRemoteRepo(ctx context.Context, css sources.ChangesetSource, targetRepo *types.Repo) (*types.Repo, error) {
+	if !conf.Get().BatchChangesEnforceForks {
+		return targetRepo, nil
+	}
+
+	fss, ok := css.(sources.ForkableChangesetSource)
+	if !ok {
+		return nil, errChangesetSourceCannotFork
+	}
```

I could see it being potentially more useful if, when encountering a changeset source which does not support forks, we fallback on just creating a branch on the target repo like before. I'm not worried about this for now considering all three of our support codehosts do support forking, but just wanted to capture the thought before I forget.

---

I think this is alright, we want to support it on all and don't want to widen the gap we already created with draft changesets on bitbucket

---

I'd prefer to fail noisily than do something potentially wrong silently, since that might result in us trying to push to an arbitrary branch on the target repo.